### PR TITLE
Enhance APIs of mux simulator and add documentation

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Enhance the mux simulator so that it would be easier to manipulate the mux status from test script side.
Add documentation.

#### How did you do it?
This change made some enhancements to the mux simulator and added documentation.
* Add support of "toggle" active side.
* Add support of randomly set active side.
* Add support of set active side for all bridges of specified vm_set.

#### How did you verify/test it?
Use curl to test the affected APIs.

### CODE REVIEW

File: ansible/roles/vm_set/files/mux_simulator.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+The mux simulator would be deployed to test server as a systemd server `mux-simulator` during `testbed-cli.sh add-topo`.
+```
+azure@test-server:~/veos-vm$ sudo systemctl status mux-simulator
+鈼 mux-simulator.service - mux simulator
+   Loaded: loaded (/etc/systemd/system/mux-simulator.service; static; vendor preset: enabled)
+   Active: active (running) since Fri 2020-12-25 07:56:21 UTC; 12s ago
+ Main PID: 928 (python)
+    Tasks: 1 (limit: 11059)
+   CGroup: /system.slice/mux-simulator.service
```
Comment (bingwang-ms) [inline ansible/roles/vm_set/files/mux_simulator.md:32]: There are some garbled character. Could you have a recheck?

File: ansible/roles/vm_set/files/mux_simulator.py [hunk 2]
```diff
@@ -268,18 +269,29 @@ def set_active_side(vm_set, port_index, new_active_side):
     Args:
         vm_set (string): The vm_set of test setup.
         port_index (int or string): Index of the port.
-        new_active_side (string): Either "tor_a" or "tor_b".
+        new_active_side (string): One of: 'tor_a', 'tor_b', 'toggle', 'random'. If new_active_side is 'toggle',
+            always toggled the active side. If new_active_side is 'random', randomly choose new side from 'tor_a'
+            and 'tor_b'.
```
Comment (bingwang-ms) [inline ansible/roles/vm_set/files/mux_simulator.py:274]: What's the purpose for toggling to a random port?

File: ansible/roles/vm_set/files/mux_simulator.py [hunk 3]
```diff
@@ -382,28 +394,53 @@ def get_mux_bridges(vm_set):
...
+        new_all_mux_status[bridge] = set_active_side(vm_set, port_index, new_active_side)
+    return new_all_mux_status
+
+
+@app.route('/mux/<vm_set>', methods=['GET', 'POST'])
 def all_mux_status(vm_set):
     """Handler for requests to /mux/<vm_set>.
 
     For GET request, return detailed status of all the mux Y cables belong to the specified vm_set.
+    For POST request, update mux active side according to poseted data.
```
Comment (bingwang-ms) [inline ansible/roles/vm_set/files/mux_simulator.py:421]: Could you give a data example for posting data to the whole vm_set? I'm not very clear. Thanks

File: ansible/roles/vm_set/files/mux_simulator.py (hunk 2 above)
Comment (wangxin) [reply ansible/roles/vm_set/files/mux_simulator.py:274]: I believe that in production, not all mux cables have the same active side. This is to simulate the possible state in production.

File: ansible/roles/vm_set/files/mux_simulator.py (hunk 3 above)
Comment (wangxin) [reply ansible/roles/vm_set/files/mux_simulator.py:421]: Sure, I'll update.

File: ansible/roles/vm_set/files/mux_simulator.md (hunk 1 above)
Comment (wangxin) [reply ansible/roles/vm_set/files/mux_simulator.md:32]: Sure, I'll update.

**Review by bingwang-ms — APPROVED** [review]

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**